kafka-rest/src/test/java/io/confluent/kafkarest/controllers/PartitionManagerImplTest.java

```diff
@@ -370,6 +371,28 @@ public void getPartition_nonExistingCluster_throwsNotFound() throws Exception {
     }
   }
 
+  @Test
+  public void getPartition_nonExistingCluster_throwsUnknownTopicOrPartition() throws Exception {
+    expect(topicManager.getTopic(CLUSTER_ID, TOPIC_NAME))
+        .andReturn(
+            failedFuture(
+                new UnknownTopicOrPartitionException(
+                    String.format(
+                        "This server does not host topic-partition %d for topic %s",
+                        0, TOPIC_NAME))));
+    replay(topicManager);
+
+    try {
+      partitionManager.getPartition(CLUSTER_ID, TOPIC_NAME, PARTITION_1.getPartitionId()).get();
```

Comment:
You can do `assertThrows` if you want to verify that an exception should be thrown. `assertThrowns` returns the thrown exception and you can verify the error message accordingly.

Reply:
Discussed offline. We want to keep the test style consistent with other tests in the same test class.